Src/Base/AMReX_Loop.H

```diff
@@ -211,6 +211,32 @@ void LoopConcurrentOnCpu (Box const& bx, int ncomp, F&& f) noexcept
     }}}}
 }
 
+// Implementation of "constexpr for" based on
+// https://artificial-mind.net/blog/2020/10/31/constexpr-for
+//
+// Approximates what one would get from a compile-time
+// unrolling of the loop
+// for (int i = 0; i < N; ++i) {
+//    f(i);
+// }
+//
+// The mechanism is recursive: we evaluate f(i) at the current
+// i and then call the for loop at i+1. f() is a lambda function
+// that provides the body of the loop and takes only an integer
+// i as its argument. It is assumed that at the loop call site
+// the lambda is declared [&] so that it can operate on the data
```

Comment:
I am not sure how what this comment about `[&]` means. If the loop body uses a pointer (a raw pointer or Array4), I don't think you need `[&]` and `[=]` would also work. (I am saying = is better.)

Reply:
That was there because of its specific usage in Microphysics. You are right that it is not helpful when copied to AMReX for general usage. I've removed that comment just to avoid confusion.